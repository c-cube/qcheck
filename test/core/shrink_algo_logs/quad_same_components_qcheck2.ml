open QCheck2

let quad_same =
  Test.make ~name:"quadruples have same components"
    Gen.(quad int int int int)
    (Log.shrinks Print.(quad int int int int) @@
     fun (h,i,j,k) -> h=i || i=j || j=k)

let rand = Random.State.make [|1234|];;
try Test.check_exn ~rand quad_same with Test.Test_fail _ -> ()
