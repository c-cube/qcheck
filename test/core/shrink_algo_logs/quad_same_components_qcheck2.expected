fails (-2548545221485712023, 3367339764960236310, -4202639126704614816, 2724675603984413065)
fails (196990738990232254, 3367339764960236310, -4202639126704614816, 2724675603984413065)
fails (0, 3367339764960236310, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1683669882480118155, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 841834941240059077, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 420917470620029538, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 210458735310014769, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 105229367655007384, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 52614683827503692, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 26307341913751846, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 13153670956875923, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 6576835478437961, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 3288417739218980, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1644208869609490, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 822104434804745, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 411052217402372, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 205526108701186, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 102763054350593, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 51381527175296, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 25690763587648, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 12845381793824, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 6422690896912, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 3211345448456, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1605672724228, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 802836362114, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 401418181057, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 200709090528, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 100354545264, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 50177272632, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 25088636316, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 12544318158, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 6272159079, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 3136079539, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1568039769, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 784019884, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 392009942, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 196004971, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 98002485, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 49001242, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 24500621, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 12250310, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 6125155, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 3062577, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1531288, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 765644, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 382822, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 191411, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 95705, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 47852, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 23926, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 11963, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 5981, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 2990, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1495, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 747, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 373, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 186, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 93, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 46, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 23, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 11, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 5, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 2, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1, -4202639126704614816, 2724675603984413065)
holds (0, 0, -4202639126704614816, 2724675603984413065)
fails (0, 1, 2425249609035529582, 2724675603984413065)
holds (0, 0, 2425249609035529582, 2724675603984413065)
fails (0, 1, 0, 2724675603984413065)
holds (0, 0, 0, 2724675603984413065)
holds (0, 1, 0, 0)
fails (0, 1, 0, 1362337801992206532)
holds (0, 0, 0, 1362337801992206532)
holds (0, 1, 0, 0)
fails (0, 1, 0, 681168900996103266)
holds (0, 0, 0, 681168900996103266)
holds (0, 1, 0, 0)
fails (0, 1, 0, 340584450498051633)
holds (0, 0, 0, 340584450498051633)
holds (0, 1, 0, 0)
fails (0, 1, 0, 170292225249025816)
holds (0, 0, 0, 170292225249025816)
holds (0, 1, 0, 0)
fails (0, 1, 0, 85146112624512908)
holds (0, 0, 0, 85146112624512908)
holds (0, 1, 0, 0)
fails (0, 1, 0, 42573056312256454)
holds (0, 0, 0, 42573056312256454)
holds (0, 1, 0, 0)
fails (0, 1, 0, 21286528156128227)
holds (0, 0, 0, 21286528156128227)
holds (0, 1, 0, 0)
fails (0, 1, 0, 10643264078064113)
holds (0, 0, 0, 10643264078064113)
holds (0, 1, 0, 0)
fails (0, 1, 0, 5321632039032056)
holds (0, 0, 0, 5321632039032056)
holds (0, 1, 0, 0)
fails (0, 1, 0, 2660816019516028)
holds (0, 0, 0, 2660816019516028)
holds (0, 1, 0, 0)
fails (0, 1, 0, 1330408009758014)
holds (0, 0, 0, 1330408009758014)
holds (0, 1, 0, 0)
fails (0, 1, 0, 665204004879007)
holds (0, 0, 0, 665204004879007)
holds (0, 1, 0, 0)
fails (0, 1, 0, 332602002439503)
holds (0, 0, 0, 332602002439503)
holds (0, 1, 0, 0)
fails (0, 1, 0, 166301001219751)
holds (0, 0, 0, 166301001219751)
holds (0, 1, 0, 0)
fails (0, 1, 0, 83150500609875)
holds (0, 0, 0, 83150500609875)
holds (0, 1, 0, 0)
fails (0, 1, 0, 41575250304937)
holds (0, 0, 0, 41575250304937)
holds (0, 1, 0, 0)
fails (0, 1, 0, 20787625152468)
holds (0, 0, 0, 20787625152468)
holds (0, 1, 0, 0)
fails (0, 1, 0, 10393812576234)
holds (0, 0, 0, 10393812576234)
holds (0, 1, 0, 0)
fails (0, 1, 0, 5196906288117)
holds (0, 0, 0, 5196906288117)
holds (0, 1, 0, 0)
fails (0, 1, 0, 2598453144058)
holds (0, 0, 0, 2598453144058)
holds (0, 1, 0, 0)
fails (0, 1, 0, 1299226572029)
holds (0, 0, 0, 1299226572029)
holds (0, 1, 0, 0)
fails (0, 1, 0, 649613286014)
holds (0, 0, 0, 649613286014)
holds (0, 1, 0, 0)
fails (0, 1, 0, 324806643007)
holds (0, 0, 0, 324806643007)
holds (0, 1, 0, 0)
fails (0, 1, 0, 162403321503)
holds (0, 0, 0, 162403321503)
holds (0, 1, 0, 0)
fails (0, 1, 0, 81201660751)
holds (0, 0, 0, 81201660751)
holds (0, 1, 0, 0)
fails (0, 1, 0, 40600830375)
holds (0, 0, 0, 40600830375)
holds (0, 1, 0, 0)
fails (0, 1, 0, 20300415187)
holds (0, 0, 0, 20300415187)
holds (0, 1, 0, 0)
fails (0, 1, 0, 10150207593)
holds (0, 0, 0, 10150207593)
holds (0, 1, 0, 0)
fails (0, 1, 0, 5075103796)
holds (0, 0, 0, 5075103796)
holds (0, 1, 0, 0)
fails (0, 1, 0, 2537551898)
holds (0, 0, 0, 2537551898)
holds (0, 1, 0, 0)
fails (0, 1, 0, 1268775949)
holds (0, 0, 0, 1268775949)
holds (0, 1, 0, 0)
fails (0, 1, 0, 634387974)
holds (0, 0, 0, 634387974)
holds (0, 1, 0, 0)
fails (0, 1, 0, 317193987)
holds (0, 0, 0, 317193987)
holds (0, 1, 0, 0)
fails (0, 1, 0, 158596993)
holds (0, 0, 0, 158596993)
holds (0, 1, 0, 0)
fails (0, 1, 0, 79298496)
holds (0, 0, 0, 79298496)
holds (0, 1, 0, 0)
fails (0, 1, 0, 39649248)
holds (0, 0, 0, 39649248)
holds (0, 1, 0, 0)
fails (0, 1, 0, 19824624)
holds (0, 0, 0, 19824624)
holds (0, 1, 0, 0)
fails (0, 1, 0, 9912312)
holds (0, 0, 0, 9912312)
holds (0, 1, 0, 0)
fails (0, 1, 0, 4956156)
holds (0, 0, 0, 4956156)
holds (0, 1, 0, 0)
fails (0, 1, 0, 2478078)
holds (0, 0, 0, 2478078)
holds (0, 1, 0, 0)
fails (0, 1, 0, 1239039)
holds (0, 0, 0, 1239039)
holds (0, 1, 0, 0)
fails (0, 1, 0, 619519)
holds (0, 0, 0, 619519)
holds (0, 1, 0, 0)
fails (0, 1, 0, 309759)
holds (0, 0, 0, 309759)
holds (0, 1, 0, 0)
fails (0, 1, 0, 154879)
holds (0, 0, 0, 154879)
holds (0, 1, 0, 0)
fails (0, 1, 0, 77439)
holds (0, 0, 0, 77439)
holds (0, 1, 0, 0)
fails (0, 1, 0, 38719)
holds (0, 0, 0, 38719)
holds (0, 1, 0, 0)
fails (0, 1, 0, 19359)
holds (0, 0, 0, 19359)
holds (0, 1, 0, 0)
fails (0, 1, 0, 9679)
holds (0, 0, 0, 9679)
holds (0, 1, 0, 0)
fails (0, 1, 0, 4839)
holds (0, 0, 0, 4839)
holds (0, 1, 0, 0)
fails (0, 1, 0, 2419)
holds (0, 0, 0, 2419)
holds (0, 1, 0, 0)
fails (0, 1, 0, 1209)
holds (0, 0, 0, 1209)
holds (0, 1, 0, 0)
fails (0, 1, 0, 604)
holds (0, 0, 0, 604)
holds (0, 1, 0, 0)
fails (0, 1, 0, 302)
holds (0, 0, 0, 302)
holds (0, 1, 0, 0)
fails (0, 1, 0, 151)
holds (0, 0, 0, 151)
holds (0, 1, 0, 0)
fails (0, 1, 0, 75)
holds (0, 0, 0, 75)
holds (0, 1, 0, 0)
fails (0, 1, 0, 37)
holds (0, 0, 0, 37)
holds (0, 1, 0, 0)
fails (0, 1, 0, 18)
holds (0, 0, 0, 18)
holds (0, 1, 0, 0)
fails (0, 1, 0, 9)
holds (0, 0, 0, 9)
holds (0, 1, 0, 0)
fails (0, 1, 0, 4)
holds (0, 0, 0, 4)
holds (0, 1, 0, 0)
fails (0, 1, 0, 2)
holds (0, 0, 0, 2)
holds (0, 1, 0, 0)
fails (0, 1, 0, 1)
holds (0, 0, 0, 1)
holds (0, 1, 0, 0)
